### DISCUSSION TITLE
Is there any way to completely reinstall zmodules defined in zimrc?

### DISCUSSION THREAD

**Discussion by fmbarina** [zimfw/zimfw#525] [Q&A] [answered]
Without having to delete directories manually, I mean.

Might've found a rare-ish bug using Git to install plugins, will open issue later if reproducible.

If I understand correctly (barely looked at the code), the first time running `zimfw install` it will clone the necessary repositories. In subsequent interactions with these repositories (e.g. when running `zimfw update`), it will only execute commands such as `git fetch` or `git pull`.

What I'm asking is if there's any way to tell zim to do a `git clone` again, as if forcefully reinstalling plugins.

**Comment by ericbn** [comment, marked as answer]
Hi @fmbarina. You're correct about your understanding of the `install` and `update` actions of `zimfw`.

Currently the only way to reinstall a module is to manually delete its `${ZIM_HOME}/modules/<module_name>` directory and then run `zimfw install` again.

We could add a new `reinstall` action that would reinstall all modules. Is this what you're trying to do?

**Comment by fmbarina** [comment]
While I'm not sure if it's necessary, I'd consider a `reinstall` command in the future regardless. Feels like something that ends up being useful sooner or later for unpredictable reasons.

And my question was badly worded, I only needed to reinstall one plugin due to an issue with it. I knew it was as easy as deleting its directory (which is what I ended up doing), but I wondered if there was a way to do this through Zim.

To clarify, I *think* the issue was that since I started using a fork of the plugin, when I finally changed its `owner/plugin` name in my zimrc to the upstream project, Zim didn't know how to handle it (error on line 720 running `update`). I imagine it saw the plugin as installed because the name remained the same, when in reality the remote didn't even exist anymore.

But I could be wrong, I'm shooting blind :grimacing:. Should I try to reproduce and open an issue?

**Reply by ericbn** [reply]
Manually reconfiguring the local repository can break how zimfw works, yeah that can happen. For example, if there's no "origin" remote, it will fail. We just try to fail fast at the first failed operation and show a meaningful error message. [Here](https://github.com/zimfw/zimfw/blob/master/src/stage2/75_zimfw_tool_git.zsh.erb)'s all that the zimfw install, check or update actions do with git.

Feel free to open an issue if you have steps to reproduce it consistently and then a clear expected behavior.
